ra/ra.go

```diff
@@ -2122,6 +2139,16 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 			"Order cannot contain more than %d DNS names", ra.maxNames)
 	}
 
+	if req.CertificateProfileName != "" {
+		pv, ok := ra.validationProfiles[req.CertificateProfileName]
```

Comment:
We should probably do something with the `if !ok` case here.

Reply:
I'm going to infer--from this comment--that if we have a mapping, and the requested profile isn't included in it, we want to treat that request as malformed. Let me know if that isn't the case.